data/core/doc/init.lua

```diff
@@ -509,6 +511,20 @@ function Doc:text_input(text, idx)
     if line1 ~= line2 or col1 ~= col2 then
       self:delete_to_cursor(sidx)
     end
+
+    if self.overwrite
+    and col1 ~= #self.lines[line1]
+    and #text:gsub('\n', '') ~= 0 then
+      -- temporary solution
+      local _, next_col = nil, col1
+      local len = text:ulen()
+      while len > 0 do
+        _, next_col = translate.next_char(self, line1, next_col)
+        len = len - 1
+      end
+      self:remove(line1, col1, line1, next_col)
+    end
```

Comment:
```suggestion
    if
      self.overwrite and #text:gsub('\n', '') ~= 0
      and
      (col1 ~= #self.lines[line1] or text:find("\n", 1, true))
    then
      local next_line, next_col = line1, col1
      local len = text:ulen()
      while len > 0 do
        next_line, next_col = translate.next_char(self, next_line, next_col)
        len = len - 1
      end
      self:remove(line1, col1, next_line, next_col)
    end
```
This adds multi-line overwrite support:

https://github.com/lite-xl/lite-xl/assets/1702572/8e7723fc-f733-46fa-81b1-729153189d6c

Reply:
In general overtype doesn't apply to copy/paste operations, nor does it continue to other lines.

Reply:
that would be weird anyways